### ISSUE TITLE
Include skipped signatures in VerificationResult

### ISSUE THREAD

**Issue by Hayden-IO** [sigstore/sigstore-go#48]
**Description**

https://github.com/sigstore/sigstore-go/pull/47 and https://github.com/sigstore/sigstore-go/pull/45 introduce skipping log and TSA signatures respectively that the trust bundle cannot verify. This information is not passed back to the verifier, the signatures are just silently skipped over.

We can update [VerificationResults](https://github.com/sigstore/sigstore-go/blob/9a59c7f51a05d2305a8ba4ab9343416aaaeb16d6/pkg/verify/signed_entity.go#L160) to pass this information back, which could be helpful for debugging that the bundle contains the expected trust root material.

**Comment by Hayden-IO** [comment]
We might also want to split out parsing errors from verification errors. For example, for timestamp verification, we will skip a timestamp either if it is an invalid RFC3161 structure or if verification fails.

**Comment by Hayden-IO** [comment]
At the sigstore-go meeting, we discussed that `VerificationResult` currently contains metadata that is verified. @phillmv pointed out that this struct could be passed to a policy controller for further evaluation. Adding a field like `UnverifiedResults` into this struct might mislead verifiers.

**Comment by dmitris** [comment]
I had a related experience and asked a question on [Slack](https://sigstore.slack.com/archives/C0440BFT43H/p1729613321322739):

```
a question about this error handling instance:
https://github.com/sigstore/sigstore-go/blob/main/pkg/verify/tsa.go#L113-L116
		// Ensure timestamp responses are from trusted sources
		timestamp, err := tsaverification.VerifyTimestampResponse(signedTimestamp, bytes.NewReader(dsseSignatureBytes), trustedRootVerificationOptions)
		if err != nil {
			continue
		}
don't we want to expose / return those errors in some form?  For example, I'm now struggling with an ASN.1 parsing error  from [github.com/digitorus/timestamp.ParseResponse](http://github.com/digitorus/timestamp.ParseResponse) (called from [github.com/sigstore/timestamp-authority/pkg/verification.VerifyTimestampResponse](http://github.com/sigstore/timestamp-authority/pkg/verification.VerifyTimestampResponse)):

error parsing response into Timestamp: asn1: structure error: tags don't match (16 vs {class:1 tag:13 length:73 isCompound:false}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} response @2

but I had to run the debugger to see this error - otherwise all that is "exposed" to the user is what seems somewhat cryptic error:
failed to verify timestamps: threshold not met for verified signed & log entry integrated timestamps: 0 < 1
maybe there could be at least some verbose or even debug flag that one can enable (or propagate the option) for easier troubleshooting and error overview, before firing up the debugger

[tsa.go](https://github.com/sigstore/sigstore-go/blob/main/pkg/verify/tsa.go)
        timestamp, err := tsaverification.VerifyTimestampResponse(signedTimestamp, bytes.NewReader(dsseSignatureBytes), trustedRootVerificationOptions)
        if err != nil {
            continue
        }
```

@haydentherapper responded there:

> We do want to surface the errors, but within each verification, we want to ignore anything that doesn't properly verify - the challenge is separating what we can't verify because it's malformed vs what we don't trust, so we just skip over any failure.

**Comment by cmurphy** [comment]
Is this still worth taking a look at?

I'm not clear on how having a new `Unverified` field would mislead policy controllers, I think if they're not currently looking for that field then they would continue to ignore it.

The problem now is with the 1.0 release it will be difficult to change some `Verify*` methods to return unverified results. `VerifySignedTimestamp` returns a list of verification errors but its caller `VerifyObserverTimestamps` buries it, and `VerifyTlogEntry`/`VerifyTransparencyLogInclusion` does not surface verification errors. We could deprecate those methods and create new ones that return the unverified entities.

**Comment by steiza** [comment]
I don't think this is a priority, compared to unblocking folks with Cosign questions / fixing Cosign gaps.
